code/modules/projectiles/gun.dm

```diff
@@ -508,6 +509,11 @@
 		//As opposed to no-delay pew pew
 		P.accuracy += 2
 
+	var/datum/firemode/F = firemodes[sel_mode]
+	if(one_hand_fa_penalty > 2 && !wielded && F?.name == "full auto") // todo: make firemode names defines
```

Comment:
As a note: the behaviour of a firemode should be dictated by flags rather than names.